### PR TITLE
[PeerDAS] rework ENR custody_subnet_count and add tests

### PR DESCRIPTION
**Pull request by 0x00101010**
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**
Rework ENR custody_subnet_count related code

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: beacon-chain/p2p/discovery.go [hunk 1]
```diff
@@ -43,6 +44,11 @@ const (
 	udp6
 )
 
+const (
+	quickProtocolEnrKey      = "quic"
```
Comment (nalepae) [inline beacon-chain/p2p/discovery.go:48]: This const is never used.
I guess you wanted to do something like:
```go
func (quicProtocol) ENRKey() string { return quickProtocolEnrKey }
```
?

File: beacon-chain/p2p/discovery.go [hunk 2]
```diff
@@ -43,6 +44,11 @@ const (
...
+	custodySubnetCountEnrKey = "custody_subnet_count"
```
Comment (nalepae) [inline beacon-chain/p2p/discovery.go:49]: I think you miss something like:

```go
func (CustodySubnetCount) ENRKey() string { return custodySubnetCountEnrKey }
```

**Comment by nalepae** [comment]
Thank you for your pull request!

The majority of your changes are scheduled for deletion because of https://github.com/ethereum/consensus-specs/pull/3772.

If you wish, you can implement this specification change into Prysm, so we'll be able to merge it once the specification PR is merged. (In the contrary, just fix the few comments I made and we'll merge this one.)

**Comment by 0x00101010** [comment]
> Thank you for your pull request!
> 
> The majority of your changes are scheduled for deletion because of [ethereum/consensus-specs#3772](https://github.com/ethereum/consensus-specs/pull/3772).
> 
> If you wish, you can implement this specification change into Prysm, so we'll be able to merge it once the specification PR is merged. (In the contrary, just fix the few comments I made and we'll merge this one.)

@nalepae updated, let me know if it looks good

**Comment by nalepae** [comment]
Thank you!

I approved it, and set it as `blocked` until https://github.com/ethereum/consensus-specs/pull/3772 is merged.

**Review by nalepae — APPROVED** [review]

**Review by nalepae — CHANGES_REQUESTED** [review]
You need to run
```
bazel run //:gazelle -- fix
```

(CI fails.)

**Review by nalepae — APPROVED** [review]

**Comment by nalepae** [comment]
Approved, thanks!

**Pull request merged**